owncloudData/src/main/java/com/owncloud/android/data/files/db/OCFileEntity.kt

```diff
@@ -0,0 +1,126 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.data.files.db
+
+import android.database.Cursor
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILES_TABLE_NAME
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_ACCOUNT_OWNER
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_CONTENT_LENGTH
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_CONTENT_TYPE
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_CREATION
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_ETAG
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_ETAG_IN_CONFLICT
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_IS_DOWNLOADING
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_KEEP_IN_SYNC
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_LAST_SYNC_DATE
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_LAST_SYNC_DATE_FOR_DATA
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_MODIFIED
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_MODIFIED_AT_LAST_SYNC_FOR_DATA
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_NAME
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_PARENT
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_PATH
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_PERMISSIONS
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_PRIVATE_LINK
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_REMOTE_ID
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_SHARED_VIA_LINK
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_SHARED_WITH_SHAREE
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_STORAGE_PATH
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_TREE_ETAG
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta.FILE_UPDATE_THUMBNAIL
+import com.owncloud.android.data.ProviderMeta.ProviderTableMeta._ID
+import com.owncloud.android.domain.ext.isOneOf
+import com.owncloud.android.domain.files.model.MIME_DIR
+import com.owncloud.android.domain.files.model.MIME_DIR_UNIX
+
+@Entity(
+    tableName = FILES_TABLE_NAME
+)
+data class OCFileEntity(
+    var parentId: Long? = null,
+    val owner: String,
+    val remotePath: String,
+    val remoteId: String?,
+    val length: Long,
+    val creationTimestamp: Long?,
+    val modificationTimestamp: Long,
+    val mimeType: String,
+    val etag: String?,
+    val permissions: String?,
+    val privateLink: String? = null,
+    val storagePath: String? = null,
+    var name: String? = null,
+    val treeEtag: String? = null,
```

Comment:
I want to anyhow indicate duplicate files. For this I need a `hash` field. 
I know that having this field on server makes most sense, but I know nothing about the server code, so this small solution, to have in only in Android client, satisfy my needs too.

It would make my fork much easier, because I don't need to handle db changes.
Any chance to add this `val hash: String? = null` field here ?

Reply:
AFAIK duplications are detected when synchronizing a file or folder in the client. 

We already have the etag and the remote id.

Why would you need an additional field to indicate duplicated files?

Reply:
Also checksums are available ownCloud server, and already used by desktop and iOS apps. But their use case is more data integrity, not so much duplicate detection.

Reply:
@abelgardep No duplicates are found when they are in different directories.
Next to this, a rename could be also a source of duplicates.
@michaelstingl 
You could use it for duplicates detection as well